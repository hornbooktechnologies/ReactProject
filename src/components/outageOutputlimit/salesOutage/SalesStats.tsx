import { useTranslation } from 'react-i18next'
import { SalesOutageCardModel } from '../../../models/outageOutputLimit/SalesOutageModel'
import { AppConstant } from '../../../utils/AppConstants'
import { getDiffOfValues, IsappearTooltip } from '../../../utils/utils'
import Tooltip from '../../common/Tooltip'
import './../../../assets/scss/outageOutputLimit/_outage.scss'

const SalesStats = (props: SalesOutageCardModel) => {
  const { i18n } = useTranslation()
  let diffPlannedPlan = getDiffOfValues(props.plannedPlan,props.plannedPreviousPlan)
  let diffPlannedActual = getDiffOfValues(props.plannedActual,props.plannedPreviousActual)
  let diffUnplannedActual = getDiffOfValues(props.unplannedActual,props.unplannedPreviousActual)

  return (
    <div className="uk-card">
      <h6 className="font-18">{props.header}</h6>
      <label className="font-14">
        {i18n.t('OUTAGE_AND_OUTPUT_LIMIT.PLANNED_PLAN')}
      </label>
      <div className="font-20 uk-text-bold uk-margin-remove mb-20">
        <span
          className={`${IsappearTooltip(diffPlannedPlan)}`}
        >
          {props.plannedPlan}
          {diffPlannedPlan === true ? (
            <Tooltip
              prefix={''}
              previousDayValue={props.plannedPreviousPlan}
              unit={AppConstant.OUTAGE_MEASURE_UNIT}
            />
          ) : null}
          <span className="font-10 uk-text-normal">
            {AppConstant.OUTAGE_MEASURE_UNIT}
          </span>
        </span>
        <span className="font-10 uk-text-normal">
          {(props.plannedPlanSubValue || props.plannedPlanSubValue === 0) && (
            <span className="font-16">
              (<span>{props.plannedPlanSubValue}</span>
              <span className="font-10">
                {i18n.t('OUTAGE_AND_OUTPUT_LIMIT.OUTAGE_PRICE_SUB_UNIT')}
              </span>)
            </span>
          )}
       </span>
      </div>
      <div
        className="uk-child-width-1-1@s uk-child-width-1-2@m uk-grid"
        uk-grid="true"
      >
        <div>
          <p className="uk-margin-remove font-14">
            {i18n.t('OUTAGE_AND_OUTPUT_LIMIT.PLANNED_ACTUAL')}
          </p>
          <div className="uk-margin-remove uk-text-bold font-20">
            <span className={`${IsappearTooltip(diffPlannedActual)}`}>
              {props.plannedActual}
              <span className="font-10 uk-text-normal">
                {AppConstant.OUTAGE_MEASURE_UNIT}
              </span>
              {diffPlannedActual === true ? (
                <Tooltip
                  prefix={''}
                  previousDayValue={props.plannedPreviousActual}
                  unit={AppConstant.OUTAGE_MEASURE_UNIT}
                />
              ) : null}
            </span>
            <span className="uk-text-normal">
              {(props.plannedActualSubValue ||
                props.plannedActualSubValue === 0) && (
                <span className="font-16">
                  (<span>{props.plannedActualSubValue}</span>
                  <span className="font-10">
                    {i18n.t('OUTAGE_AND_OUTPUT_LIMIT.OUTAGE_PRICE_SUB_UNIT')}
                  </span>
                  )
                </span>
              )}
            </span>
          </div>
        </div>
        <div>
          <p className="uk-margin-remove font-14">
            {i18n.t('OUTAGE_AND_OUTPUT_LIMIT.UNPLANNED_ACTUAL')}
          </p>
          <div className="uk-margin-remove uk-text-bold font-20">
            <span className={`${IsappearTooltip(diffUnplannedActual)}`} >
              {props.unplannedActual}
              <span className="font-10 uk-text-normal">
                {AppConstant.OUTAGE_MEASURE_UNIT}
              </span>
              {diffUnplannedActual === true ? (
                <Tooltip
                  prefix={''}
                  previousDayValue={ props.unplannedPreviousActual}
                  unit={AppConstant.OUTAGE_MEASURE_UNIT}
                />
              ) : null}
            </span>
            <span className="font-10 uk-text-normal">
              {(props.unplannedActualSubValue ||
                props.unplannedActualSubValue === 0) && (
                <span className="font-16">
                 (<span>{props.unplannedActualSubValue}</span>
                 <span className="font-10">
                   {i18n.t('OUTAGE_AND_OUTPUT_LIMIT.OUTAGE_PRICE_SUB_UNIT')}
                 </span>
                 )
               </span>
              )}
            </span>
          </div>
        </div>
      </div>
    </div>
  )
}

export default SalesStats
