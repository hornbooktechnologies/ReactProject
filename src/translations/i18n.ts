import { initReactI18next } from 'react-i18next'
import i18n from 'i18next'
import LanguageDetector from 'i18next-browser-languagedetector'
import { TRANSLATIONS_EN } from './en/Translations'
import { TRANSLATIONS_JA } from './ja/Translations'

i18n
  .use(LanguageDetector)
  .use(initReactI18next)
  .init({
    lng: 'en',
    resources: {
      en: {
        translation: TRANSLATIONS_EN,
      },
      ja: {
        translation: TRANSLATIONS_JA,
      },
    },
  })

export default i18n
