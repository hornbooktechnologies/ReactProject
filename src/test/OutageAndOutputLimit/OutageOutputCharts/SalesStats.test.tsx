import React from "react";
import renderer from "react-test-renderer";
import SalesStats from "../../../components/outageOutputlimit/salesOutage/SalesStats";
import i18n from "../../i18nForTests";
import { I18nextProvider } from "react-i18next";

describe("Component: SalesStats", () => {
  it("SalesStats renders correctly", () => {
    const salesTestData = {
        plannedPlan: 8,
        plannedActual: 2,
        unplannedActual: 4,
        plannedActualSubValue: 3,
        unplannedActualSubValue: 2.2,
        plannedPlanSubValue: 1.2,
        header: 'Planned'
    }
    const tree = renderer
      .create(
        <I18nextProvider i18n={i18n}>
          <SalesStats {...salesTestData} />
        </I18nextProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
