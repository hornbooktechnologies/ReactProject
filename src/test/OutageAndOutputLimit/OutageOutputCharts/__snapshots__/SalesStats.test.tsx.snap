// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: SalesStats SalesStats renders correctly 1`] = `
<div
  className="uk-card"
>
  <h6
    className="font-18"
  >
    Planned
  </h6>
  <label
    className="font-14"
  >
    OUTAGE_AND_OUTPUT_LIMIT.PLANNED_PLAN
  </label>
  <div
    className="font-20 uk-text-bold uk-margin-remove mb-20"
  >
    <span
      className=""
    >
      8
      <span
        className="font-10 uk-text-normal"
      >
        円/kWh
      </span>
    </span>
    <span
      className="font-10 uk-text-normal"
    >
      <span
        className="font-16"
      >
        (
        <span>
          1.2
        </span>
        <span
          className="font-10"
        >
          OUTAGE_AND_OUTPUT_LIMIT.OUTAGE_PRICE_SUB_UNIT
        </span>
        )
      </span>
    </span>
  </div>
  <div
    className="uk-child-width-1-1@s uk-child-width-1-2@m uk-grid"
    uk-grid="true"
  >
    <div>
      <p
        className="uk-margin-remove font-14"
      >
        OUTAGE_AND_OUTPUT_LIMIT.PLANNED_ACTUAL
      </p>
      <div
        className="uk-margin-remove uk-text-bold font-20"
      >
        <span
          className=""
        >
          2
          <span
            className="font-10 uk-text-normal"
          >
            円/kWh
          </span>
        </span>
        <span
          className="uk-text-normal"
        >
          <span
            className="font-16"
          >
            (
            <span>
              3
            </span>
            <span
              className="font-10"
            >
              OUTAGE_AND_OUTPUT_LIMIT.OUTAGE_PRICE_SUB_UNIT
            </span>
            )
          </span>
        </span>
      </div>
    </div>
    <div>
      <p
        className="uk-margin-remove font-14"
      >
        OUTAGE_AND_OUTPUT_LIMIT.UNPLANNED_ACTUAL
      </p>
      <div
        className="uk-margin-remove uk-text-bold font-20"
      >
        <span
          className=""
        >
          4
          <span
            className="font-10 uk-text-normal"
          >
            円/kWh
          </span>
        </span>
        <span
          className="font-10 uk-text-normal"
        >
          <span
            className="font-16"
          >
            (
            <span>
              2.2
            </span>
            <span
              className="font-10"
            >
              OUTAGE_AND_OUTPUT_LIMIT.OUTAGE_PRICE_SUB_UNIT
            </span>
            )
          </span>
        </span>
      </div>
    </div>
  </div>
</div>
`;
